src/nanoFramework.IoT.Device.CodeConverter/GenericsTemplates/Span.cs

```diff
@@ -0,0 +1,186 @@
+        /// </summary>
+        /// <param name="destination"> The destination System.Span object.</param>
+        /// <exception cref="System.ArgumentException">
+        /// destination is shorter than the source System.Span.
+        /// </exception>
+        public void CopyTo(Span<T> destination)
+        {
+            if (destination.Length < _length)
+            {
+                throw new ArgumentException();
+            }
+
+            for (int i = 0; i < _length; i++)
+            {
+                destination[i] = _array[_start + i];
+            }
+        }
+
+        /// <summary>
+        /// Forms a slice out of the current span that begins at a specified index.
+        /// </summary>
+        /// <param name="start">The index at which to begin the slice.</param>
+        /// <returns>A span that consists of all elements of the current span from start to the end of the span.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">start is less than zero or greater than System.Span.Length.</exception>
+        public Span<T> Slice(int start)
+        {
+            return Slice(start, _length - start);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the current span starting at a specified index for a specified length.
+        /// </summary>
+        /// <param name="start">The index at which to begin this slice.</param>
+        /// <param name="length">The desired length for the slice.</param>
+        /// <returns>A span that consists of length elements from the current span starting at start.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">start or start + length is less than zero or greater than System.Span.Length.</exception>
+        public Span<T> Slice(int start, int length)
+        {
+            if ((start < 0) || (length < 0) || (start + length > _length))
+            {
+                // start or start + length is less than zero or greater than length
+                throw new ArgumentOutOfRangeException();
+            }
+
+            return new Span<T>(_array, _start + start, length);
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into a new array.
+        /// </summary>
+        /// <returns> An array containing the data in the current span.</returns>
+        public T[] ToArray()
+        {
+            var array = new T[_length];
+            for (int i = 0; i < _length; i++)
+            {
+                array[i] = _array[_start + i];
+            }
+
+            return array;
+        }
+
+        public static implicit operator Span<T>(T[] array)
+        {
+            return new(array);
+        }
+
+        /// <summary>
+        /// Defines an implicit conversion of a System.Span to a System.ReadOnlySpan.
+        /// </summary>
+        /// <param name="span">The object to convert to a System.ReadOnlySpan.</param>
+        public static implicit operator ReadOnlySpan<T>(Span<T> span)
```

Comment:
remove this one, we don't need ReadOnlySpan, we've simplify and get read of the ReadOnlySpan to only have Span